### PR TITLE
Guillaume

### PR DESCRIPTION
**Pull request by LPogi-FR**
PR before merge with the branch master.

### CODE REVIEW

File: com/hemebiotech/analytics/WriteSymptomDataToFile.java [hunk 1]
```diff
@@ -2,7 +2,33 @@
...
+* Anything that will read symptom data from a source;
+* The important part is, the return value from the operation, which is a list of strings,
+* that may contain many duplications.
+* The implementation does not need to order the list.
+*/
+public class WriteSymptomDataToFile implements ISymptomWriter {
+  /** 
+   Attribute with the name of the output File.
+  */
+  private final String fileOutput = "result.out";
```
Comment (metaXIII) [inline com/hemebiotech/analytics/WriteSymptomDataToFile.java:13]: no need for javadoc for this variables, javadoc is supposed to be existing only for method

File: com/hemebiotech/analytics/ISymptomWriter.java [hunk 2]
```diff
@@ -2,11 +2,10 @@
...
-
-/**
- * Anything the will write symptom data on a valid data output format
+/**.
+ * Anything that will write symptom data on a valid data output format
  *
  */
 public interface ISymptomWriter {
-// WRITE YOUR CODE HERE
+  public void writeSymptoms(Map<String, Integer> symptoms);
```
Comment (metaXIII) [inline com/hemebiotech/analytics/ISymptomWriter.java:14]: missing javadoc

File: com/hemebiotech/analytics/AnalyticsCounter.java [hunk 3]
```diff
@@ -1,43 +1,59 @@
...
+
+  /**
+  * Method that @return the same map sorted by Key. 
+  * @param symptoms a {@link Map} with the number of occurrences for each symptom.
+  */
+  public Map<String, Integer> sortSymptoms(Map<String, Integer> symptoms) {
+    return new TreeMap<>(symptoms);
+  }
+
+  public void writeSymptoms(Map<String, Integer> symptoms) {
```
Comment (metaXIII) [inline com/hemebiotech/analytics/AnalyticsCounter.java:59]: by consistency, do not hesitate to create javadoc here also

File: com/hemebiotech/analytics/ReadSymptomDataFromFile.java [hunk 4]
```diff
@@ -6,42 +6,38 @@
...
-			} catch (IOException e) {
-				e.printStackTrace();
-			}
-		}
-		
-		return result;
-	}
+  private String filepath;
+  /**.
+   *
```
Comment (metaXIII) [inline com/hemebiotech/analytics/ReadSymptomDataFromFile.java]: missing global description (be careful with your dot on line 16)

File: com/hemebiotech/analytics/WriteSymptomDataToFile.java [hunk 5]
```diff
@@ -2,7 +2,33 @@
 
 import java.io.FileWriter;
 import java.io.IOException;
-import java.util.Iterator;
 import java.util.Map;
 
-// WRITE YOUR CODE HERE
\ No newline at end of file
+
+/**.
```
Comment (metaXIII) [inline com/hemebiotech/analytics/WriteSymptomDataToFile.java]: no need for the dot here on line 8

**Review by metaXIII — CHANGES_REQUESTED** [review]
Do not hesitate to include .gitignore in order to avoid the .class available at each run

**Comment by LPogi-FR** [comment]
> 

Just modified what have been asked.

File: com/hemebiotech/analytics/AnalyticsCounter.java (hunk 3 above)
Comment (metaXIII) [reply com/hemebiotech/analytics/AnalyticsCounter.java:59]: missing `@param symptoms Map<String, Integer> symptoms`

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+*.class
+symptoms.txt
```
Comment (metaXIII) [inline .gitignore]: in order to test the program, include symptoms.txt to the PR

**Pull request merged**